core/src/main/java/org/akaza/openclinica/service/crfdata/EnketoUrlService.java

```diff
@@ -199,13 +217,22 @@ public String getEditUrl(String subjectContextKey, PFormCacheSubjectContextEntry
         // Return Enketo URL
         List<FormLayoutMedia> mediaList = formLayoutMediaDao.findByEventCrfId(eventCrf.getEventCrfId());
         EnketoURLResponse eur = enketo.registerAndGetEditURL(formLayout, crfFlavor, populatedInstance, subjectContextKey, redirectUrl, markComplete, studyOid, mediaList,
-                goTo, flavor);
+                goTo, flavor, role, study, studyEvent, mode);
```

Comment:
Is there a way to simply this? There are just way too many parameters to this method.
Maybe we can create a RegisterEnketo object?